docs/scenarios/jdk_8_to_17_migration.md

```diff
@@ -0,0 +1,395 @@
+# Modernizing a Java 8 Application to Java 17 Using Konveyor AI
+
+## Goal
+
+This scenario demonstrates how **Konveyor AI** can assist in modernizing a **Java 8** application to **Java 17** by identifying outdated APIs, suggesting migration changes, and ensuring best security practices (e.g., replacing MD5 hashing). We will use a custom rule and default rules with **Konveyor AI** to analyze a legacy Java application and generate refactoring recommendations.
+
+## Prerequisites
+
+Ensure you have the following set up:
+
+- [VSCode](https://code.visualstudio.com/download)
+- [Git](https://git-scm.com/downloads)
+- [GenAI credentials](https://github.com/konveyor/kai/blob/main/docs/llm_selection.md#openai-service)
+- Java 8 & Java 17 installed
+- Maven 3.9+
+- [Java sample app](https://github.com/konveyor-ecosystem/migrationex.git)
+- [Kai VSCode IDE extension 0.0.13+](https://github.com/konveyor/editor-extensions/releases)
+
+## Application Overview
+
+The [`migration example`](<(https://github.com/konveyor-ecosystem/migrationex.git)>) repository consists of two main components:
+
+1.  Java 8 Application – The sample Java 8 application (`migration example`) is a simple **Book management system** that manages a collection of books.
+2.  Rules Folder – Contains custom migration rules used by Kai for this app.
+
+### Files to modernize
+
+- `Library.java`
+- `LibraryService.java`
+- `pom.xml`
+
+## Step 1: Setup the environment
+
+### 1.1 Clone the Sample Java 8 Application
+
+Clone the migration example repository:
+
+```sh
+
+git clone https://github.com/konveyor-ecosystem/migrationex.git
+cd migrationex/migrationex
+```
+
+Navigate to File > Open in VSCode and locate the folder we just cloned and select the sub-folder `migrationex`.
+
+### 1.2 Install the Konveyor AI VS Code Extension
+
+Follow this [installation guide](https://github.com/konveyor/kai/blob/main/docs/scenarios/demo.md#configure-konveyor) to install the latest Konveyor AI IDE plugin.
+
+_Note: At the time of writing this guide, we used version 0.0.13._
+
+### 1.3 Configure Kai in VS Code
+
+1.3.1. When you launch the extension, you will land on the Welcome Page, as shown below. If the Welcome Page does not appear, proceed to the step 1.3.2.
+
+![walkthrough-1](../images/walkthrough-1.png)
+
+1.3.2. If "Set up Konveyor" is not available in the list, click the More button for additional options.
+![walkthrough-2](../images/walkthrough-2.png)
+
+1.3.3 If the welcome page does not appear, open the command palette by pressing Command + Shift + P. This will bring up a list of commands.
+![walkthrough-3](../images/walkthrough-3.png)
```

Comment:
Consider if we want to remove this screenshot, I found it a little confusing to see this screenshot, I had to stop for a few seconds and think of why I was seeing a screen that almost looked like a duplicate.   I think we'd be OK with keeping the text with the hint for Command + Shift + P.